lib/export_vacancy_records_to_big_query.rb

```diff
@@ -0,0 +1,118 @@
+require 'google/cloud/bigquery'
+
+class ExportVacancyRecordsToBigQuery
+  attr_reader :dataset
+
+  def initialize(bigquery: Google::Cloud::Bigquery.new)
+    @dataset = bigquery.dataset ENV.fetch('BIG_QUERY_DATASET')
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def run!(batch_size: 1_000)
+    Vacancy.all.find_in_batches batch_size: batch_size do |batch|
+      dataset.insert 'vacancies', present_for_big_query(batch), autocreate: true do |schema|
+        schema.string 'id', mode: :required
+        schema.string 'slug', mode: :required
+        schema.string 'job_title', mode: :required
+        schema.string 'minimum_salary', mode: :required
+        schema.string 'maximum_salary', mode: :nullable
+        schema.date 'starts_on', mode: :nullable
+        schema.date 'ends_on', mode: :nullable
+        schema.string 'subjects', mode: :repeated
+
+        schema.string 'min_pay_scale', mode: :nullable
+        schema.string 'max_pay_scale', mode: :nullable
+        schema.string 'leadership', mode: :nullable
+
+        schema.string 'education', mode: :nullable
+        schema.string 'qualifications', mode: :nullable
+        schema.string 'experience', mode: :nullable
+        schema.string 'status', mode: :required
+
+        schema.timestamp 'expiry_time', mode: :nullable
+        schema.timestamp 'publish_on', mode: :nullable
+
+        schema.record 'school', mode: :nullable do |school|
+          school.string 'urn', mode: :required
+          school.string 'county', mode: :required
+        end
+
+        schema.timestamp 'created_at', mode: :required
+        schema.timestamp 'updated_at', mode: :required
+
+        schema.string 'application_link', mode: :nullable
+
+        schema.boolean 'newly_qualified_teacher', mode: :required
+
+        schema.integer 'total_pageviews', mode: :nullable
+        schema.integer 'total_get_more_info_clicks', mode: :nullable
+        schema.string 'working_patterns', mode: :repeated
+
+        schema.string 'listed_elsewhere', mode: :nullable
+        schema.string 'hired_status', mode: :nullable
+        schema.boolean 'pro_rata_salary', mode: :nullable
+
+        schema.string 'publisher_user_id', mode: :nullable
+      end
+    end
+  end
+  # rubocop:enable Metrics/AbcSize
+
+
+  private
+
+  # TODO: Check what the last updated line and append new data after that.
```

Comment:
Did you mean to leave this in?

Reply:
Not really. Will remove this comment and add a JIRA ticket instead.
Thanks!